File: Python/Strategy/ducks/strategy/decoy_duck.py
from duck import *
from quack_behaviour import *
from fly_behaviour import *

class DecoyDuck(Duck):

    def __init__(self):
        fly_no_way_instance = FlyNoWay()
        self.set_fly_behaviour(fly_no_way_instance)
        mute_instance = MuteQuack()
        self.set_quack_behaviour(mute_instance)

    def display(self):
        print "I'm a decoy duck"

File: Cpp/hfdp/src/strategy/DecoyDuck.cpp
//
// Created by Johnson, Chase on 12/4/16.
//

#include <iostream>
#include "DecoyDuck.h"
#include "FlyNoWay.h"
#include "MuteQuack.h"


DecoyDuck::DecoyDuck() {
    this->setFlyBehaviour(new FlyNoWay());
    this->setQuackBehaviour(new MuteQuack());
}

void DecoyDuck::display() {
    std::cout << "I'm a duck decoy" << std::endl;
}


